src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -815,6 +831,23 @@ public SolutionState WithRunAnalyzers(ProjectId projectId, bool runAnalyzers)
             return ForkProject(newProject);
         }
 
+        /// <summary>
+        /// Create a new solution instance with the project specified updated to have
+        /// the specified attributes, except for <see cref="ProjectInfo.ProjectAttributes.Version"/> which is auto-incremented.
+        /// </summary>
+        public SolutionState WithProjectAttributes(ProjectId projectId, ProjectInfo.ProjectAttributes attributes, bool updateDocuments)
+        {
+            var oldProject = GetRequiredProjectState(projectId);
+            var newProject = oldProject.WithAttributes(attributes, updateDocuments);
+
+            if (oldProject == newProject)
+            {
+                return this;
+            }
+
+            return ForkProject(newProject);
```

Comment:
So this method shares the same bug as WithProjectChecksumAlgorithm if updateDocuments is true, since the documents and their trees are now out of sync with the compilation we have in hand. Per earlier comments, if we want to keep this method (I don't think we should), just remove the updateDocuments parameter, and then add a long comment that we're letting things get out of sync, but since we aren't actually changing document states we don't need to replace trees.

Reply:
Removed updateDocuments parameter